tests/test_orchestrator_comprehensive.py

```
@@ -0,0 +1,243 @@
+    def test_should_download_release(self, orchestrator):
+        """Test determining if a release should be downloaded."""
+        release = Release(tag_name="v2.7.14", prerelease=False)
+        result = orchestrator._should_download_release(release, "firmware")
+        assert isinstance(result, bool)
+
+    def test_handle_download_result(self, orchestrator):
+        """Test handling download results."""
+        # Create a mock download result
+        mock_result = Mock()
+        mock_result.success = True
+        mock_result.file_path = "/tmp/test/file.bin"
```

Ruff (0.14.8)

Probable insecure usage of temporary file or directory: "/tmp/test/file.bin"

(S108)